inference-engine/src/legacy_api/src/ie_cnn_layer_builder_ngraph.cpp

```diff
@@ -1168,7 +1168,7 @@ CNNLayer::Ptr NodeConverter<ngraph::op::v1::Split>::createLayer(const std::share
     if (!axis_node_const) {
         THROW_IE_EXCEPTION << "Split " << castedLayer->get_friendly_name() << " has no axes as Constant";
     }
-    auto axis = axis_node_const->get_vector<int64_t>()[0];
+    auto axis = axis_node_const->cast_vector<int64_t>()[0];
```

Comment:
Why don't you use `get_data_ptr()`? This method is faster and doesn't create new vectors.

Reply:
According to spec it can be any integer type and using get_data_ptr() you will cast values manually. get_vector will make a copy and may raise an exception here, so the solution is cast_vector.
Also this inputs contains axes and it's not so huge as convolution weights.

Reply:
I think you can use [this method](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/src/ngraph/op/constant.hpp#L426) in order to don't cast values manually :)

Reply:
How it can be used if it can have any integer type? write giant switch for any integer type? 
axis_node_const->get_data_ptr<int64_t>() will raise exception if types do not match.

Reply:
Oh yeah, you are right. Thanks!